2013-09-19  Raphael Amiard  <amiard@adacore.com>

   * gdk/win32/gdkdevice-win32.c (gdk_device_win32_query_state) : Apply 
   _gdk_offset_x/y offsets to the root_x/y variables so that they become
   absolute relative to the origin in  the virtual desktop space rather than 
   relative to the main monitor

--- gdk/win32/gdkdevice-win32.orig.c	2013-09-20 14:38:56.083718500 +0200
+++ gdk/win32/gdkdevice-win32.c	2013-09-20 14:39:50.373823700 +0200
@@ -187,10 +187,10 @@ gdk_device_win32_query_state (GdkDevice
   GetCursorPos (&point);
 
   if (root_x)
-    *root_x = point.x;
+    *root_x = point.x + _gdk_offset_x;
 
   if (root_y)
-    *root_y = point.y;
+    *root_y = point.y + _gdk_offset_y;
 
   ScreenToClient (hwnd, &point);

2013-09-19  Raphael Amiard  <amiard@adacore.com>

   * gtk/gtkmenu.c (gtk_menu_position) : If a position func was supplied,
     always re-get the monitor after it has been applied to the coordinates

--- gtk/gtkmenu.c.orig	2013-09-19 17:49:38.172727600 +0200
+++ gtk/gtkmenu.c	2013-08-30 13:08:29.000000000 +0200
@@ -4601,8 +4601,7 @@ gtk_menu_position (GtkMenu  *menu,
       (* priv->position_func) (menu, &x, &y, &priv->initially_pushed_in,
                                priv->position_func_data);
 
-      if (priv->monitor_num < 0)
-        priv->monitor_num = gdk_screen_get_monitor_at_point (screen, x, y);
+      priv->monitor_num = gdk_screen_get_monitor_at_point (screen, x, y);
 
       gdk_screen_get_monitor_workarea (screen, priv->monitor_num, &monitor);
     }

2012-04-20	Nicolas Setton  <setton@adacore.com>

	* gtk/gtkapplicationwindow.c: GDesktopApplication is not defined under
	Windows, do not use it.

	To be submitted.

2011-11-17	Nicolas Setton  <setton@adacore.com>

	* gdk/win32/gdkdnd-win32.c (resolve_link): Deactivate resolving of
	dnd links for now, since CLSID_ShellLink and IID_IShellLinkW are
	not found at link time for an unknown reason.
	To be investigated.

--- gdk/win32/gdkdnd-win32.c.orig	2011-11-17 14:59:18.578125000 +0100
+++ gdk/win32/gdkdnd-win32.c	2011-11-17 15:42:47.875000000 +0100
@@ -1342,6 +1342,7 @@ resolve_link (HWND     hWnd,
    * assumed that CoInitialize has been called.
    */
 
+#if 0
   hr = CoCreateInstance (&CLSID_ShellLink,
 			 NULL,
 			 CLSCTX_INPROC_SERVER,
@@ -1388,6 +1389,8 @@ resolve_link (HWND     hWnd,
     pslW->lpVtbl->Release (pslW);
 
   return SUCCEEDED (hr);
+#endif
+  return FALSE;
 }
 
 #if 0
--- gtk/gtkapplicationwindow.c.orig	2012-04-20 15:26:50.203250000 +0200
+++ gtk/gtkapplicationwindow.c	2012-04-20 15:30:48.234500000 +0200
@@ -273,6 +273,7 @@
     }
   else
     {
+#ifdef HAVE_GIO_UNIX
       /* the shell does not show it, so make sure we show it */
       if (g_menu_model_get_n_items (G_MENU_MODEL (window->priv->app_menu_section)) == 0)
         {
@@ -311,6 +312,7 @@
               g_clear_object (&app_info);
             }
         }
+#endif
     }
 }
 
